osu.Game/Online/Chat/ChannelManager.cs

```diff
@@ -445,6 +446,27 @@ protected override Task Poll()
             return tcs.Task;
         }
 
+        /// <summary>
+        /// Marks the <paramref name="channel"/> as read
+        /// <see>(see <paramref name="message"/> for more information)</see>
+        /// </summary>
+        /// <param name="channel">The channel that will be marked as read</param>
+        /// <param name="message">The message where it will be read up to. If <see langword="null"/> was provided, the latest message of the <paramref name="channel"/> will be read.</param>
+        public void MarkChannelAsRead(Channel channel, Message message = null)
+        {
+            if (message == null)
+                message = channel.Messages.Last();
+            else
+                Debug.Assert(channel.Id == message.Id, "Provided channel and message owner channel aren't equal.");
```

Comment:
i highly doubt this path is tested (the comparison is wrong). if it's not tested/used just remove it and remove the message parameter for now.

Reply:
I'm blind, I notice this way too late

Reply:
Resolved with https://github.com/ppy/osu/pull/7425/commits/2ea1367a881cab6e5e9eed787971a6f5ca22f989